### PR TITLE
Update dust-tt extension

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

- Add support for Oauth connexion flow.
- Add Command to Select Workspace (oauth only)
- Add Command for Claude-3
- Use official Dust Client (removed custom implementation)
- Support dust favorite assistants feature (oauth only, removed the localStorage implementation)
- Fix icons for referenced documents
- Fix labels for assistant scopes
- Update wording to match Dust (especially agent -> assistant).
- Improve UX (thinking animation, abort fetching when exiting view...)

## Screencast

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/dust-tt/package.json [hunk 1]
```diff
@@ -8,52 +8,104 @@
...
+    },
+    {
+      "name": "apiUrl",
+      "description": "Dust Api URL (advanced)",
       "type": "textfield",
-      "required": false,
-      "title": "API URL",
-      "default": "https://dust.tt/api/v1/w",
-      "placeholder": "https://dust.tt/api/v1/w"
+      "required": true,
```
Comment (Fraggle) [inline extensions/dust-tt/package.json:40]: I'm not 100% sure what happen if required if false but there is a default value ?

(I don't **need** to show it to the user but I need a value)

**Comment by stang** [comment]
@Fraggle — When testing this locally: the Oauth flow worked as expected. However, the assistant's answers are not displayed correctly. 

I get a `Request was aborted` error (see screenshot).

Note: the conversation has been correctly created because the `Continue On Dust` link works 👌 

image

FWIW, I'm using the latest version of `raycast v1.85.1`.

**Comment by stang** [comment]
FTR, here is the snippet of commands to run to test the extension locally, based on [this guide](https://developers.raycast.com/basics/review-pullrequest):

```bash
BRANCH="ext/dust-tt"
FORK_URL="https://github.com/Fraggle/raycast-extensions.git"
EXTENSION_NAME="dust-tt"

git clone -n --depth=1 --filter=tree:0 -b ${BRANCH} ${FORK_URL}
cd raycast-extensions
git sparse-checkout set --no-cone "extensions/${EXTENSION_NAME}"
git checkout
cd "extensions/${EXTENSION_NAME}"
npm install && npm run dev
```

**Comment by Fraggle** [comment]
@stang you need to check production in the node development mode otherwise effect are triggered twice.

This will not happen for users.

[more info](https://dev.to/jherr/react-18-useeffect-double-call-for-apis-emergency-fix-27ee)

**Comment by stang** [comment]
> @stang you need to check production in the node development mode otherwise effect are triggered twice.
> 
> This will not happen for users.
> 
> [more info](https://dev.to/jherr/react-18-useeffect-double-call-for-apis-emergency-fix-27ee)

👁️ you mean use `NODE_ENV=production`, or something more specific?

Note: here is the error log I got:

```
16:43:34.291 {
  url: 'https://dust.tt/api/v1/w/<wId>/assistant/conversations/<cId>/messages/<mId>/events',
  duration: 0,
  connectorsError: {
    type: 'unexpected_network_error',
    message: 'Unexpected network error from DustAPI: AbortError: This operation was aborted'
  },
  error: DOMException [AbortError]: This operation was aborted
      at fetch3 (/Users/stang/.config/raycast/extensions/dust-tt/ask.js:17722:17)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
} DustAPI error
```

**Comment by Fraggle** [comment]
Nope, within raycast preference. ([more info](https://raycastcommunity.slack.com/archives/C02HEMAF2SJ/p1730994818982219))

image

**Comment by stang** [comment]
LGTM!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**